daemon/main.go

```diff
@@ -219,50 +219,78 @@ func checkMinRequirements() {
+	flags.BoolVar(&config.IPv4Disabled,
+		"disable-ipv4", false, "Disable IPv4 mode")
+	flags.StringVarP(&config.DockerEndpoint,
+		"docker", "e", "unix:///var/run/docker.sock", "Register a listener for docker events on the given endpoint")
+	flags.StringVar(&config.EnablePolicy,
+		"enable-policy", endpoint.DefaultEnforcement, "Enable policy enforcement")
+	flags.BoolVar(&enableTracing,
+		"enable-tracing", false, "Enable tracing while determining policy")
+	flags.StringVar(&v4Prefix,
+		"ipv4-range", "", "IPv4 prefix for endpoints, must be in correct format (10.N.0.1/16)")
+	flags.StringVar(&config.K8sEndpoint,
+		"k8s-api-server", "", "Kubernetes api address server (for https use kubeconfig instead)")
+	flags.StringVar(&config.K8sCfgPath,
+		"k8s-kubeconfig-path", "", "Absolute path of the kubernetes kubeconfig file")
+	flags.BoolVar(&config.KeepConfig,
+		"keep-config", false, "When restoring state, keeps containers' configuration in place")
+	flags.BoolVar(&config.KeepTemplates,
+		"keep-templates", false, "Do not restore BPF template files from binary")
+	flags.StringVar(&kvStore,
+		"kvstore", "", "Key-value store type")
+	flags.Var(option.NewNamedMapOptions("kvstore-opts", &kvStoreOpts, nil),
+		"kvstore-opt", "Key-value store options")
+	flags.StringVar(&labelPrefixFile,
+		"label-prefix-file", "", "Valid label prefixes file path")
+	flags.StringSliceVar(&validLabels,
+		"labels", []string{}, "List of label prefixes used to determine identity of an endpoint")
+	flags.StringVar(&config.LBInterface,
+		"lb", "", "Enables load balancer mode where load balancer bpf program is attached to the given interface")
+	flags.StringVar(&config.LibDir,
+		"lib-dir", defaults.LibraryPath, "Directory path to store runtime build environment")
+	flags.StringSliceVar(&loggers,
+		"log-driver", []string{}, "Logging endpoints to use for example systemd, fluentd")
+	flags.Var(option.NewNamedMapOptions("log-opts", &logOpts, nil),
+		"log-opt", "Log driver options for cilium")
+	flags.BoolVar(&enableLogstash,
+		"logstash", false, "Enable logstash integration")
+	flags.StringVar(&logstashAddr,
+		"logstash-agent", "127.0.0.1:8080", "Logstash agent address")
+	flags.Uint32Var(&logstashProbeTimer,
+		"logstash-probe-timer", 10, "Logstash probe timer (seconds)")
+	flags.StringVar(&nat46prefix,
+		"nat46-range", nodeaddress.DefaultNAT46Prefix, "IPv6 prefix to map IPv4 addresses to")
+	flags.StringVarP(&v6Address,
+		"node-address", "n", "", "IPv6 address of node, must be in correct format (XXXX:XXXX:XXXX:XXXX:XXXX:XXXX::/96)")
+	flags.BoolVar(&config.RestoreState,
+		"restore", false, "Restores state, if possible, from previous daemon")
+	flags.StringVar(&socketPath,
+		"socket-path", defaults.SockPath, "Sets daemon's socket path to listen for connections")
+	flags.StringVar(&config.RunDir,
+		"state-dir", defaults.RuntimePath, "Directory path to store runtime state")
+	flags.StringVarP(&config.Tunnel,
+		"tunnel", "t", "vxlan", "Tunnel mode VXLAN or GENEVE")
```

Comment:
I believe "direct" mode is only set if the `snoop-device != "undefined"`.
For example, if you set `snoop-device="eth0"` and `-t vxlan` the `"vxlan"` is ignored.
The code is this one: https://github.com/cilium/cilium/blob/9c6885a3a4b81c2355c6981b4126f3a58c3eae1d/daemon/daemon.go#L315